sklearn/cluster/_birch.py

```diff
@@ -521,6 +525,12 @@ def fit(self, X, y=None):
         self
             Fitted estimator.
         """
+        if self.copy != "deprecated":
+            warnings.warn(
+                "`copy` was deprecated in 1.6 and will be removed in 1.8. Simply "
+                "remove this parameter as it has no effect.",
```

Comment:
```suggestion
                "`copy` was deprecated in 1.6 and will be removed in 1.8 since it"
                " has no effect internally. Simply leave this parameter to it's"
                " default value to avoid this warning."
```